cypress/integration/add_rating.spec.js

```diff
@@ -0,0 +1,27 @@
+describe(`Add A Rating`, () => {
+    it(`Should Fail Due to Missing Fields`, () => {
+        cy.visit('http://localhost:3199/user/8', {timeout: 30000})
+        cy.get(`button`, {timeout: 10000}).contains('Recipes').click()
+        cy.get(`button`, {timeout: 10000}).contains('View').click()
+        cy.get(`#submitRating`, {timeout: 10000}).click()
+        cy.on('window:alert', (text) => {
+            expect(text).to.contains('Error: Both difficulty and stars are required');
+        });
+    })
+
+    it(`Should Add A Rating`, () => {
+        const rnd = Math.floor(Math.random() * 100000)
+        cy.intercept(`/api/rating`).as('postAttempt')
+        cy.visit('http://localhost:3199/user/8', {timeout: 30000})
+        cy.get(`button`, {timeout: 10000}).contains('Recipes').click()
+        cy.get(`button`, {timeout: 10000}).contains('View').click()
+        cy.get(`#difficultyStars`, {timeout: 10000}).click()
+        cy.get(`#ratingStars`, {timeout: 10000}).click()
+        cy.get('#reviewText', {timeout: 10000}).type(`${rnd}testreview :-)`)
+        cy.get(`#submitRating`, {timeout: 10000}).click()
+        cy.wait('@postAttempt', {timeout: 10000}).then((interception) => {
+            assert.equal(interception.response.statusCode, 200, `post successful`)
+        })
+        cy.get('div', {timeout: 10000}).contains(`${rnd}testreview :-)`)
```

Comment:
The last line, which searches for a div containing testReview, needs some kind of assertion. Otherwise the contains() method returns what is found, which could be nothing, so it doesn't actually test whether the matching DOM element was created or not.

Reply:
![image](https://user-images.githubusercontent.com/64811274/162870211-7eadd90d-323e-4a13-b0e9-c6d0b7d8d4a2.png)